src/adapter/src/coord/sequencer/inner/peek.rs

```diff
@@ -639,43 +771,73 @@ impl Coordinator {
-                        if let peek::PeekPlan::FastPath(plan) = global_lir_plan.peek_plan() {
-                            let arity = global_lir_plan.typ().arity();
-                            let finishing = if !optimizer.finishing().is_trivial(arity) {
-                                Some(optimizer.finishing())
-                            } else {
-                                None
-                            };
-                            used_indexes = plan.used_indexes(finishing);
+                        match optimizer.as_mut() {
+                            // Optimize SELECT statement.
+                            Either::Left(optimizer) => {
+                                let global_mir_plan = global_mir_plan.unwrap_left();
+
+                                // Collect the list of indexes used by the dataflow at this point
+                                let mut used_indexes = {
+                                    let df_desc = global_mir_plan.df_desc();
+                                    let df_meta = global_mir_plan.df_meta();
+                                    UsedIndexes::new(
+                                    df_desc
+                                        .index_imports
+                                        .iter()
+                                        .map(|(id, _index_import)| {
+                                            (*id, df_meta.index_usage_types.get(id).expect("prune_and_annotate_dataflow_index_imports should have been called already").clone())
+                                        })
+                                        .collect(),
+                                )
+                                };
+
+                                // MIR ⇒ LIR lowering and LIR ⇒ LIR optimization (global)
+                                let global_mir_plan = global_mir_plan
+                                    .resolve(timestamp_context.clone(), ctx.session_mut());
+                                let global_lir_plan =
+                                    optimizer.catch_unwind_optimize(global_mir_plan)?;
+
+                                // Trace the resulting plan for the top-level `optimize` path.
+                                match global_lir_plan.peek_plan() {
+                                    peek::PeekPlan::FastPath(plan) => {
+                                        let arity = global_lir_plan.typ().arity();
+                                        let finishing = if !optimizer.finishing().is_trivial(arity)
+                                        {
+                                            Some(optimizer.finishing().clone())
+                                        } else {
+                                            None
+                                        };
+                                        used_indexes = plan.used_indexes(finishing.as_ref());
+
+                                        // TODO(aalexandrov): rework `OptimizerTrace` with support
```

Comment:
Note that this is adding back things that are no longer in `main`:

https://github.com/MaterializeInc/materialize/blob/783400d213914ba591c765d67667c735a6568532/src/adapter/src/coord/sequencer/inner/peek.rs#L657-L673

Can you please align the changes so the `Either::Left(optimizer)` branch is identical to what is currently in `main`?

Reply:
ack

Reply:
I was actually behind main on my branch on an earlier commit, just pulled and rebased again. So, should be good.